src/components/ec/ucc_ec.c

```diff
@@ -28,6 +30,7 @@ ucc_status_t ucc_ec_init(const ucc_ec_params_t *ec_params)
     ucc_status_t   status;
     ucc_ec_attr_t  attr;
 
+    pthread_mutex_lock(&ucc_ec_mutex);
```

Comment:
returned from this func w/o unlocking mutex in some error cases

Reply:
fixed